features/collections_dir.feature

```diff
@@ -0,0 +1,156 @@
+Feature: Collections Directory
+  As a hacker who likes to structure content without clutter
+  I want to be able to organize my collections under a single directory
+  And render them from there
+
+  Scenario: Custom collections_dir containing only posts
+    And I have a collections/_posts directory
+    And I have the following post within the "collections" directory:
+      | title         | date       | content         |
+      | Gathered Post | 2009-03-27 | Random Content. |
+    And I have a "_config.yml" file with content:
+    """
+    collections_dir: collections
+    """
+    When I run jekyll build
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And I should see "Random Content." in "_site/2009/03/27/gathered-post.html"
+
+  Scenario: Rendered collection in custom collections_dir also containing posts
+    Given I have a collections/_puppies directory
+    And I have the following document under the "puppies" collection within the "collections" directory:
+      | title  | date       | content            |
+      | Rover  | 2007-12-31 | content for Rover. |
+    And I have a collections/_posts directory
+    And I have the following post within the "collections" directory:
+      | title         | date       | content         |
+      | Gathered Post | 2009-03-27 | Random Content. |
+    And I have a "_config.yml" file with content:
+    """
+    collections:
+      puppies:
+        output: true
+
+    collections_dir: collections
+    """
+    When I run jekyll build
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And the "_site/puppies/rover.html" file should exist
+    And I should see "Random Content." in "_site/2009/03/27/gathered-post.html"
+
+  Scenario: Rendered collection in custom collections_dir with posts at the site root
+    Given I have a collections/_puppies directory
+    And I have the following document under the "puppies" collection within the "collections" directory:
+      | title  | date       | content            |
+      | Rover  | 2007-12-31 | content for Rover. |
+    And I have a _posts directory
+    And I have the following post:
+      | title        | date       | content         |
+      | Post At Root | 2009-03-27 | Random Content. |
+    And I have a "_config.yml" file with content:
+    """
+    collections:
+      puppies:
+        output: true
+
+    collections_dir: collections
+    """
+    When I run jekyll build
+    Then I should get a zero exit status
+    And the _site directory should exist
+    And the "_site/puppies/rover.html" file should exist
+    And the "_site/2009/03/27/post-at-root.html" file should not exist
```

Comment:
Woof, this is going to screw people up. We can't enforce that the `_posts` directory is in the `collections_dir` until 4.0, if we even want to.

Reply:
its not.. we just released this feature in `3.7.0` .. The sooner we ship this patch, the lesser no.of users gonna have issues..